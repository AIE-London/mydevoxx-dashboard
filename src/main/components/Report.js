import React, { Component } from 'react';
import { Row, Col } from 'react-flexbox-grid';
import TalkCard from './TalkCard';
import ReportStats from './ReportStats';


import testImage from '../../test/snapshot/images/test-image.jpeg';

const talkDetail = {
  title: 'Welcome to Devoxx 2017',
  description: 'Join the organisers of Devoxx UK and great keynote ' +
  'speakers for inspring stories in 20 minute segments.',
  rating: 4,
  topTracks: [
    'Java',
    'Devoxx',
    'Spring',
  ],
  notes: 'Lorem ipsum dolor sit amet, everti quaestio mel ea. Ex eos ' +
  'volutpat qualisque. Sale tantas cotidieque quo ut, ad nostro consectetuer' +
  ' nec. Feugiat qualisque quo an. Labores officiis te nam.',
  review: {
    name: 'Test User',
    comment: 'Great session, thanks for organising. Looking forward to the next one!',
    image: testImage
  }
};


let reportStatsData =
    {
        "minutes": 455,
        "talks": 10,
        "learning": "Spring, Java" ,
        "attendees": 435
    };

class Report extends Component {

  render() {

    let {minutes, talks, learning, attendees} = reportStatsData;

    return (
        <div className="Report">

            <Row center="xs">
                <Col xs={10}>
                    <ReportStats minutes={minutes} talks={talks} learning={learning} attendees={attendees} ></ReportStats>
                </Col>
            </Row>
          <div style={{height: '72px'}}>
          </div>
          <Row center="xs">
            <Col xs={10}>
              <TalkCard talk={talkDetail}></TalkCard>
            </Col>
          </Row>
        </div>
    );
  }
}

export default Report;
