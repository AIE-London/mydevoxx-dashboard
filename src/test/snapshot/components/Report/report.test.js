/**
 * Created by dan on 07/04/2017.
 */
import React from 'react';
import ReactDOM from 'react-dom';
import Report from '../../../../main/components/Report';
import renderer from 'react-test-renderer';

import testImage from '../../images/test-image.jpeg';

const routeData = {
    reportStats: {
        minutes: 455,
        talks: 10,
        learning: "Spring, Java",
        attendees: 435
    },
    talk: {
        dayNo: 'One',
        sTime: '10:00',
        room: 'Mezzanine',
        title: 'Welcome to Devoxx 2017',
        description: 'Join the organisers of Devoxx UK and great keynote ' +
        'speakers for inspring stories in 20 minute segments.',
        rating: 4,
        topTracks: [
            'Java',
            'Devoxx',
            'Spring',
        ],
        notes: 'Lorem ipsum dolor sit amet, everti quaestio mel ea. Ex eos ' +
        'volutpat qualisque. Sale tantas cotidieque quo ut, ad nostro consectetuer' +
        ' nec. Feugiat qualisque quo an. Labores officiis te nam.',
        review: {
            name: 'Test User',
            comment: 'Great session, thanks for organising. Looking forward to the next one!',
            image: testImage
        },
        speakers: [
            {
                name: 'Test Speaker',
                company: 'Capgemini',
                blog: 'personalblog.com',
                talks: [
                    'Intro to Devoxx (Room 1 - 11:45)',
                    'Intro to Devoxx 2 (Room 2 - 13:45)'
                ]
            }
        ]
    }
}

test('Report component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Report route={routeData}/>, div);
    });

});


test('Report component snapshot', () => {
    const tree = renderer.create(
        <Report route={routeData}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
