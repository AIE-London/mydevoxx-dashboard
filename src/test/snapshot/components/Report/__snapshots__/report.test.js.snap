exports[`test Report component snapshot 1`] = `
<div
  className="Report">
  <div
    className="row center-xs">
    <div
      className="col-xs-10">
      <div
        className="row center-xs">
        <div
          className="col-xs-10">
          <div
            className="col-xs-2">
            <div
              className="row start-xs">
              <h1
                className="iscJSK">
                MyReport
              </h1>
            </div>
          </div>
          <div
            className="row center-xs">
            <div
              className="col-xs-10">
              <h2
                className="eUofwH">
                I spent 
                <span
                  className="kiuWfQ">
                  455
                   minutes
                </span>
              </h2>
              <p
                className="jlawfx">
                Attending 
                <span
                  className="kiuWfQ">
                  10
                   talks
                </span>
              </p>
              <h2
                className="eUofwH">
                Learning what\'s new in...
              </h2>
              <p
                className="jlawfx">
                Spring, Java
                 
              </p>
              <h2
                className="eUofwH">
                Networking with
              </h2>
              <p
                className="jlawfx">
                 
                <span
                  className="kiuWfQ">
                  435
                </span>
                 Attendees
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "height": "72px",
      }
    } />
  <div
    className="row center-xs">
    <div
      className="col-xs-10">
      <div
        className="bvnlUe row start-xs">
        <div
          className="col-xs-6 fteZLe">
          <div
            className="row start-xs">
            <h2
              className="Axdp"
              id="title">
              Welcome to Devoxx 2017
            </h2>
          </div>
          <div
            className="row end-xs">
            <div
              className="row"
              left="xs">
              <div
                className="kmDnAn" />
              <div
                className="kmDnAn" />
              <div
                className="kmDnAn" />
              <div
                className="kmDnAn" />
              <div
                className="cWcebC" />
            </div>
          </div>
          <p
            className="jLJUnn">
            Join the organisers of Devoxx UK and great keynote speakers for inspring stories in 20 minute segments.
          </p>
          <div
            className="row">
            <h3
              className="jJURiy">
              Top Tracks
            </h3>
          </div>
          <div
            className="row">
            <ul
              className="izGhAk">
              <li
                className="cLQKgI">
                Java
              </li>
              <li
                className="cLQKgI">
                Devoxx
              </li>
              <li
                className="cLQKgI">
                Spring
              </li>
            </ul>
          </div>
        </div>
        <div
          className="col-xs-6 eZvXOT">
          <div
            className="row start-xs">
            <h3
              className="jJURiy"
              id="notesHeader">
              My Notes
            </h3>
          </div>
          <div
            className="row start-xs">
            <p
              className="bKHqKP">
              Lorem ipsum dolor sit amet, everti quaestio mel ea. Ex eos volutpat qualisque. Sale tantas cotidieque quo ut, ad nostro consectetuer nec. Feugiat qualisque quo an. Labores officiis te nam.
            </p>
          </div>
          <div
            className="row start-xs">
            <h3
              className="jJURiy"
              id="reviewHeader">
              My Reviews
            </h3>
          </div>
          <div
            className="fEZGiH row start-xs">
            <div
              className="row start-xs">
              <div
                className="">
                <img
                  className="imhmHM"
                  src="test-image.jpeg" />
              </div>
              <div
                className="col-xs">
                <h3
                  className="jPwRym">
                  Test User
                </h3>
                <p
                  className="ZxTsc">
                  Great session, thanks for organising. Looking forward to the next one!
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-10">
      <div
        className="lluTUT row">
        <div
          className="kblate">
          <div
            className="row">
            <div
              className="">
              <img
                className="fOjFzA"
                src="defaultIcon.png" />
            </div>
            <div
              className="col-xs">
              <div>
                <h3>
                  Test Speaker
                </h3>
                <p>
                  Capgemini
                </p>
              </div>
            </div>
          </div>
          <div
            className="row">
            <div
              className="">
              <div>
                Personal Blog:
              </div>
              <div
                className="fYLUow">
                personalblog.com
              </div>
            </div>
          </div>
          <div
            className="row">
            <div
              className="">
              <div>
                Other Talks:
              </div>
              <ul>
                <div
                  className="fYLUow">
                  <li>
                    Intro to Devoxx (Room 1 - 11:45)
                  </li>
                  <li>
                    Intro to Devoxx 2 (Room 2 - 13:45)
                  </li>
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
